### PR TITLE
GTEST/UCP: disable HWTM protov2

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
 + skip `test_ucp_tag_offload_multi` with proto_v2
 + del code duplication

## Why ?
similar to https://github.com/openucx/ucx/pull/8770

### CODE REVIEW

File: test/gtest/ucp/test_ucp_tag_match.cc [hunk 1]
```diff
@@ -37,9 +37,8 @@ class test_ucp_tag_match : public test_ucp_tag {
         if (use_proto()) {
             modify_config("PROTO_ENABLE", "y");
             modify_config("MAX_EAGER_LANES", "2");
-        } else if (RUNNING_ON_VALGRIND && m_ucp_config->ctx.proto_enable) {
-            UCS_TEST_SKIP_R("FIXME: skip forced UCX_PROTO_ENABLE=y because "
-                            "it does not completely support HWTM");
+        } else if (RUNNING_ON_VALGRIND) {
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_tag_match.cc:40]: 1. seems condition of "else" is changed
2. do we want to skip protov2 only with valgrind?

Comment (yosefe) [reply test/gtest/ucp/test_ucp_tag_match.cc]: why need to skip protov2 only on valgrind in some tests?

Comment (evgeny-leksikov) [reply test/gtest/ucp/test_ucp_tag_match.cc]: see description to https://github.com/openucx/ucx/pull/8756

File: test/gtest/ucp/test_ucp_tag_offload.cc [hunk 2]
```diff
@@ -452,6 +452,8 @@ class test_ucp_tag_offload_multi : public test_ucp_tag_offload {
 
     void init()
     {
+        skip_external_protov2();
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_tag_offload.cc:455]: why not skip only on valgrind here as well?

Comment (evgeny-leksikov) [reply test/gtest/ucp/test_ucp_tag_offload.cc:455]: this issue is different:
```
[----------] 1 test from all_rcdc/test_ucp_tag_offload_multi
[ RUN      ] all_rcdc/test_ucp_tag_offload_multi.recv_from_multi/0 <rc,dc>
/scrap/azure/agent-04/AZP_WORKSPACE/1/s/contrib/../test/gtest/ucp/test_ucp_tag_offload.cc:537: Failure
Expected equality of these values:
  0u
    Which is: 0
  ((&receiver().worker()->tm.offload.tag_hash)->size)
    Which is: 1
/scrap/azure/agent-04/AZP_WORKSPACE/1/s/contrib/../test/gtest/ucp/test_ucp_tag_offload.cc:543: Failure
Expected equality of these values:
  1u
    Which is: 1
  ((&receiver().worker()->tm.offload.tag_hash)->size)
    Which is: 2
[  FAILED  ] all_rcdc/test_ucp_tag_offload_multi.recv_from_multi/0, where GetParam() = rc,dc (1141 ms)
```
https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=56508&view=logs&j=ac2aca1a-83f8-51b1-d752-3894f913b563&t=62a8fd62-132e-5bf7-b7fb-5edbee804b9b&l=14025

**Review by yosefe — APPROVED** [review]

File: test/gtest/ucp/test_ucp_tag.h [hunk 3]
```diff
@@ -125,6 +125,10 @@ class test_ucp_tag : public ucp_test {
 
     virtual bool is_external_request();
 
+    void skip_external_protov2();
```
Comment (gleon99) [inline test/gtest/ucp/test_ucp_tag.h:128]: Minor: please apply `const` in both methods.

**Review by yosefe — APPROVED** [review]

**Pull request merged**